homeassistant/components/derivative/sensor.py

```diff
@@ -0,0 +1,177 @@
+"""Numeric derivative of data coming from a source sensor over time."""
+from decimal import Decimal, DecimalException
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_UNIT_OF_MEASUREMENT,
+    CONF_NAME,
+    STATE_UNAVAILABLE,
+    STATE_UNKNOWN,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.restore_state import RestoreEntity
+
+# mypy: allow-untyped-defs, no-check-untyped-defs
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_SOURCE_ID = "source"
+
+CONF_SOURCE_SENSOR = "source"
+CONF_ROUND_DIGITS = "round"
+CONF_UNIT_PREFIX = "unit_prefix"
+CONF_UNIT_TIME = "unit_time"
+CONF_UNIT_OF_MEASUREMENT = "unit"
```

Comment:
Import this constant from const.py

Reply:
@springstan I will import CONF_UNIT_OF_MEASUREMENT from const.py
Should I move other const declarations to const.py ?

What about the following ones, that may feel generic also :

```
# SI Metric prefixes
UNIT_PREFIXES = {None: 1, "k": 10 ** 3, "G": 10 ** 6, "T": 10 ** 9}

# SI Time prefixes
UNIT_TIME = {"s": 1, "min": 60, "h": 60 * 60, "d": 24 * 60 * 60}
```

Reply:
I would recommend that you leave them locally and not move them to const.py, since they are not used by any other integration.